### ISSUE TITLE
Unused types / imports are not being detected in *.d.ts files

### ISSUE THREAD

**Issue by massimonewsuk** [palantir/tslint#3922]
### Bug Report

- __TSLint version__: 5.10.0
- __TypeScript version__:
- __Running TSLint via__: CLI

#### TypeScript code being linted

Inside a `.d.ts` file
```ts
type Unused = {
    unused: string;
}
export type Foo = {
    foo: string;
}
```

with `tslint.json` configuration:

```json
{
  "extends": ["tslint-config-airbnb", "tslint-config-prettier"],
  "rules": {
    "import-name": [false],
    "prefer-array-literal": [false],
    "variable-name": [true, "check-format", "allow-leading-underscore"],
    "object-shorthand-properties-first": [false],
    "only-arrow-functions": [true],
    "ter-arrow-body-style": [true, "as-needed"]
  }
}
```

#### Actual behavior
No errors. The same code inside a normal *.ts file results in an error.

#### Expected behavior
I expect it to say that the Unused type is unused.

**Comment by giladgray** [comment]
@massimonewsuk are you running `tslint --project ...`? must use the project approach as these unused rules require the type checker.
https://palantir.github.io/tslint/rules/no-unused-variable/

**Comment by massimonewsuk** [comment]
Yep I'm using `tslint --project`

As mentioned above, this only affects *.d.ts files

**Comment by JoshuaKGoldberg** [comment]
Duplicate of #3883.
